packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -378,6 +383,9 @@ class _ModalScope extends StatefulWidget {
 }
 
 class _ModalScopeState extends State<_ModalScope> {
+  /// See scopedWillPopCallback in ModalRoute.
```

Comment:
the route callback mechanism should be a list of callbacks, not a single callback